ui/js/controllers/perf/e10s-trend.js

```diff
@@ -0,0 +1,462 @@
+                // scope (*sigh*) and I don't know of any other workaround
+                $scope.selectedRepo = selectedRepo;
+                $scope.selectedBaseDate = selectedBaseDate;
+                $scope.selectedNewDate = selectedNewDate;
+                $scope.selectedSampleSize = selectedSampleSize;
+                updateURL();
+                loadData();
+            };
+
+            loadData();
+        });
+    }
+
+]);
+
+perf.controller('e10sTrendSubtestCtrl', [
+    '$state', '$stateParams', '$scope', '$rootScope', '$q', '$http', '$httpParamSerializer',
+    'ThRepositoryModel', 'ThResultSetModel', 'PhSeries', 'PhCompare',
+    'thServiceDomain', 'thDefaultRepo', 'phTimeRanges', 'defaultSampleSize',
+    'phDatePoints', 'defaultBaseDate', 'defaultNewDate', 'phBlockers',
+    function e10sTrendCtrl($state, $stateParams, $scope, $rootScope, $q, $http, $httpParamSerializer,
+                      ThRepositoryModel, ThResultSetModel, PhSeries, PhCompare,
+                      thServiceDomain, thDefaultRepo, phTimeRanges,
+                      defaultSampleSize, phDatePoints, defaultBaseDate, defaultNewDate) {
+
+        var baseSignature = $stateParams.baseSignature;
+        var e10sSignature = $stateParams.e10sSignature;
+
+        $scope.compareResults = {};
+        $scope.titles = {};
+
+        $scope.sampleSizes= phTimeRanges;
+        $scope.datePoints = phDatePoints;
+        $scope.selectedSampleSize = _.find($scope.sampleSizes, {
+            value: ($stateParams.timerange) ? parseInt($stateParams.timerange) : defaultSampleSize
+        });
+        $scope.selectedBaseDate = _.find($scope.datePoints, {
+            value: ($stateParams.basedate) ? parseInt($stateParams.basedate) : defaultBaseDate
+        });
+        $scope.selectedNewDate = _.find($scope.datePoints, {
+            value: ($stateParams.newdate) ? parseInt($stateParams.newdate) : defaultNewDate
+        });
+        $scope.revision = $stateParams.revision;
+        $scope.dataLoading = true;
+
+        function displayResults(baseResults, newResults) {
+            var match = {};
+
+            $scope.compareResults = {};
+
+            // create one object of common results
+            Object.keys(baseResults).forEach(function(baseTestName) {
+                baseResults[baseTestName].forEach(function(baseResult) {
+                    if (newResults[baseTestName]) {
+                        var newResult = newResults[baseTestName];
+                        newResult = _.find(newResult, function(obj) { return obj.name === baseResult['name']; });
+                        if (match) {
+                            var trendResult = PhCompare.getTrendMap(baseTestName, baseResult, newResult);
+                            if (!$scope.compareResults[baseTestName]) {
+                                $scope.compareResults[baseTestName] = [{'baseResult': baseResult, 'newResult': newResult, 'trendResult': trendResult}];
```

Comment:
If you're creating an object you don't need to put quotes around the properties

i.e. you could just do `{baseResult: baseResult, ... }`

Also, you can avoid a bit of duplication by making the if clause just initialize the array if it doesn't exist, then pushing unconditionally:

```js
if (!$scope.compareResults[baseTestName]) {
   $scope.compareResults[baseTestName] = [];
}
$scope.compareResults[baseTestName].push({baseResult: baseResult, ...});
```